api/src/services/users/users.js

```diff
@@ -6,3 +6,45 @@ export const getCustomerId = async ({ id }) => {
     where: { id: parseInt(id) },
   })
 }
+
+export const getUserByCustomerId = async ({ customerId }) => {
+  return await db.user.findUnique({
+    where: {
+      customerId: customerId,
+    },
+    select: {
+      name: true,
+      email: true,
+    },
+  })
+}
+
+export const updateUserByCustomerId = async ({ customerId, payload }) => {
+  return await db.user.update({
+    where: {
+      customerId: customerId,
+    },
+    data: payload,
+  })
+}
+
+// update db if name and email has changed
+export const handleDBSync = async (customerId, nextName, nextEmail) => {
+  const { name, email } = await getUserByCustomerId({
+    customerId: customerId,
+  })
+  if (nextEmail !== email || nextName !== name) {
+    const payload = {}
+    if (nextEmail !== email) {
+      payload.email = nextEmail
+    }
+    if (nextName !== name) {
+      payload.name = nextName
+    }
+    return await updateUserByCustomerId({
+      customerId: customerId,
+      payload: payload,
+    })
+  }
+  return
+}
```

Comment:
Sometimes the code's a little more readable if you return early; there's just less nesting I think is the thing. I'll add a commit to demonstrate

Reply:
Here's the commit that hopefully illustrates what I mean https://github.com/redwoodjs/example-store/pull/96/commits/b802da65f2c8961ce30fe2b5d32bf87495dd1488

Reply:
@jtoar I see what you mean. Yeah that makes sense. Thanks!